examples/addons/websocket-inject-message.py

```diff
@@ -0,0 +1,31 @@
+"""
+Inject a WebSocket message into a running connection.
+
+This example shows how to inject a WebSocket message into a running connection.
+"""
+import asyncio
+
+from mitmproxy import ctx, http
+
+
+# Simple example: Inject a message as a response to an event
+
+def websocket_message(flow):
+    last_message = flow.websocket.messages[-1]
+    if b"secret" in last_message.content:
+        last_message.kill()
+        ctx.master.commands.call("inject", [flow], not last_message.from_client, "ssssssh")
+
+
+# Complex example: Schedule a periodic timer
+
+async def inject_async(flow: http.HTTPFlow):
+    msg = "hello from mitmproxy! "
+    while flow.server_conn.connected:
```

Comment:
I know this is just an example add-on, but does this work? I think it does? My thoughts go in a direction of `keep-alive` etc. But WebSockets are always HTTP 1 and I assume once it's upgraded the connection will only be used for the WebSocket. But I think it gets recycled if the WebSocket is closed and another connection to the same host is needed?

`WebSocketData` has three properties that concern `close` but isn't it lacking the actual information if the WebSocket is still open?

Reply:
There is no recycling, but your overall point is good. Next to the close properties we have, I'll add a `websocket.timestamp_end`.